crates/rome_js_analyze/src/utils/escape.rs

```diff
@@ -0,0 +1,85 @@
+pub fn escape<'a>(
+    unescaped_string: &'a str,
+    needs_escaping: &[&str],
+    escaping_char: char,
+) -> std::borrow::Cow<'a, str> {
+    let mut escaped = String::new();
+    let escaping_char_len = escaping_char.len_utf8();
+    let mut iter = unescaped_string.char_indices();
+    let mut escape_count = 0;
+    'unescaped: while let Some((idx, chr)) = iter.next() {
+        if chr == escaping_char {
+            escape_count += 1;
+        }
```

Comment:
The `escape_count` can be simplified to a `bool` because all we need to know is if we have seen an escape character before. 

```suggestion
        if chr == escaping_char {
           is_escaped = !is_escaped;
        }
```


I think it should even allow to simplify the `escape_count % 2 != 0`  by doing

```rust
  if chr == escaping_char {
	is_escaped = !is_escaped;
	continue;
  }

  for candidate in needs_escaping {
    if unescaped_string[idex..].starts_with(candidate) {
    	if !is_escaped {
			if escaped.is_empty() {
				...
			}
			...
		} else {
		}
    }
  }
```

Reply:
I thought that we need to know how many escape character is there before characters needs escaping. If there are an odd number of escape characters, a new escape is necessary, but if there are an even number of escape characters, there is no need to add an escape character.

ex)

unescaped: `\${}`     escaped: `\${}`     → no need
unescaped: `\\${}`.   escaped: `\\\${}`.  → need to add `\`
unescaped: `\\\${}`.  escaped: `\\\${}`.  → no need

Reply:
I don't think we need the count (as `count % 2` shows). What we need to know is if it is escaped or not. 

* `\\\${` -> has one escaped `\` and one escaped `${` It's not that the escape count is 3 (to my understanding).

Reply:
I understand 👍 I refactored and simplified the logic.